### PR TITLE
Fix UPDATE where the SET expression contains a qualified name.

### LINKED ISSUE #2369: sql: UPDATE .. SET fails when set expression contains qualified name
`UPDATE kv SET v = v + 1` fails with the error `qualified name "v" not found`. Are handling of the UPDATE expressions is simplistic and needs to resolve qualified names and update their value on each row being updated.

### PR DESCRIPTION
**Pull request by petermattis**
We now use SELECT to render the update values for each row. This allows
statements like "UPDATE t SET v = v + 1" to work.

Fixes #2369.

### CODE REVIEW

File: sql/update.go [hunk 1]
```diff
@@ -129,9 +123,11 @@ func (p *planner) Update(n *parser.Update) (planNode, error) {
 		// Compute the new secondary index key:value pairs for this row.
 		//
 		// Update the row values.
+		newVals := rowVals[len(tableDesc.Columns):]
 		for i, col := range cols {
-			val := vals[i]
-
+			// Our updated value expressions occur immediately after the plain
```
Comment (tamird) [inline sql/update.go]: this comment really belongs on L126

**Comment by tamird** [comment]
Nice. LGTM

File: sql/update.go (hunk 1 above)
Comment (petermattis) [inline sql/update.go]: Yep. Done.

**Comment by petermattis** [comment]
`TestStoreRangeMergeNonConsecutive` is very flaky for me on CircleCI and completely unrelated to this change.

**Pull request merged**